tests/jballerina-unit-test/src/test/resources/test-src/isolation-analysis/isolation_analysis_negative.bal

```diff
@@ -330,3 +330,59 @@ object {} testInvalidAccessOfSelfAsCapturedVariableInIsolatedFunction = object {
         self.words.filter(
             isolated function (string word) returns boolean => self.length == word.length()).length();
 };
+
```

Comment:
Shouldn't below be allowed
```ballerina
isolated function dummyTest() {
    var x = [2, 4, 6, 8].iterator().next;
    _ = x(); //@error invalid invocation of a non-isolated function in an 'isolated' function

    _ = [2, 4, 6, 8].iterator().next(); //allowed
}
```

Reply:
A bound method is `isolated` only if the object AND the method are `isolated`. So the type inferred for `x` doesn't have isolated.

https://ballerina.io/spec/lang/master/#field-access-expr

> If T is a subtype of the object basic type, then T must have a member with name field-name. The field-access-expr is evaluated by first evaluating the expression to get a value obj. If the member is a field, then the result of the expression if the value of that field of obj and the static type of the field-access-expr is the type of that field of T. If that member is a method, then the result of the expression is a new function value that when called will call the method with self bound to obj; **the isolated bit of the new function value is set if and only if the isolated bits of both obj and the method are set;** the type descriptor of the new function value will have the same annotations as the type descriptor of the method.

An isolated method of a non-isolated object can access mutable fields of the object in an unsafe way.